src/redux/charts.ts

```diff
@@ -0,0 +1,235 @@
+ */
+interface ChartsUpdateChartTypeAction {
+  type: typeof CHARTS_UPDATE_CHART_TYPE;
+  payload: ChartType;
+  secondStore: boolean;
+}
+
+/**
+ * The action for updating chart data.
+ */
+interface ChartsUpdateAction {
+  type: typeof CHARTS_UPDATE;
+  payload: ChartsState['charts'];
+}
+
+/**
+ * The action for updating the ETH-USD day chart data.
+ */
+interface ChartsUpdateUsdDayAction {
+  type: typeof CHARTS_UPDATE_USD_DAY;
+  payload: ChartsState['chartsEthUSDDay'];
+}
+
+/**
+ * The action for updating the ETH-USD month chart data.
+ */
+interface ChartsUpdateUsdMonthAction {
+  type: typeof CHARTS_UPDATE_USD_MONTH;
+  payload: ChartsState['chartsEthUSDMonth'];
+}
+
+/**
+ * A message received from a chart data provider.
+ */
+interface ChartsReceivedMessage {
+  meta?: {
+    charts_type?: ChartType;
+    currency?: string;
+  };
+  payload: {
+    charts: {
+      [key: string]: [number, number][];
+    };
+  };
+}
+
+/**
+ * Updates the current chart type in state.
+ *
+ * @param chartType The new chart type.
+ * @param secondStore Whether or not this change should update the second chart
+ * store.
+ */
+export const chartsUpdateChartType = (
+  chartType: ChartType,
+  secondStore: boolean
+) => (dispatch: Dispatch<ChartsUpdateChartTypeAction>) =>
+  dispatch({
+    payload: chartType,
+    secondStore,
+    type: CHARTS_UPDATE_CHART_TYPE,
+  });
+
+/**
+ * Updates charts in state to reflect new data.
+ *
+ * @param message The `ChartsReceivedMessage`.
+ */
+export const assetChartsReceived = (message: ChartsReceivedMessage) => (
+  dispatch: Dispatch<
+    ChartsUpdateAction | ChartsUpdateUsdDayAction | ChartsUpdateUsdMonthAction
+  >,
+  getState: AppGetState
+) => {
+  const chartType = get(message, 'meta.charts_type');
+  const { charts: existingCharts } = getState().charts;
+  const assetCharts = get(message, 'payload.charts', {});
+  const { nativeCurrency } = getState().settings;
+  if (toLower(nativeCurrency) === message?.meta?.currency) {
+    const newChartData = mapValues(assetCharts, (chartData, address) => ({
+      ...existingCharts[address],
+      // .slice to prevent mutation
+      [chartType]: reverse(chartData?.slice()),
+    }));
+    const updatedCharts = {
+      ...existingCharts,
+      ...newChartData,
+    };
+    dispatch({
+      payload: updatedCharts,
+      type: CHARTS_UPDATE,
+    });
+  }
+
+  if (
+    message?.meta?.currency === currenyTypes.usd &&
+    assetCharts[ETH_ADDRESS]
+  ) {
+    if (message?.meta?.charts_type === 'm') {
+      dispatch({
+        payload: reverse(assetCharts[ETH_ADDRESS]),
+        type: CHARTS_UPDATE_USD_MONTH,
+      });
+    } else if (message?.meta?.charts_type === 'd') {
+      dispatch({
+        payload: reverse(assetCharts[ETH_ADDRESS]),
+        type: CHARTS_UPDATE_USD_DAY,
+      });
+    }
+  }
+};
+
+// -- Reducer ----------------------------------------- //
+
+const INITIAL_STATE: ChartsState = {
+  charts: {},
+  chartsEthUSDDay: {},
```

Comment:
just realized this is an incorrect default - should be `[]`
same for line 197 `chartsEthUSDMonth`

Reply:
@jinchung Should be set to `[]` by default now for both, thanks!